include/streaming_ordered_output_thread_pool_executor.hpp

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2022 Transit Live Mapping Solutions
+ * All rights reserved.
+ *
+ * Authors:
+ *   Marenz Schmidl
+ *   Tassilo Tanneberger
+ */
+
+#pragma once
+
+#include <condition_variable>
+#include <deque>
+#include <functional>
+#include <map>
+#include <mutex>
+#include <thread>
+#include <type_traits>
+#include <vector>
+
+// thread pool executing work but outputting it the order of the input
+template <typename ReturnType> class StreamingOrderedOutputThreadPoolExecutor {
+
+  public:
+    StreamingOrderedOutputThreadPoolExecutor(int numWorkers);
```

Comment:
maybe considerung std::size_t instead of int. And argument names should be in sneak case